### PR TITLE
Better training set rebalancing

### LINKED ISSUE #533: Add better class rebalancing to LogisticRegression

### PR DESCRIPTION
**Pull request by henryre**
See `LabelBalancer`

cc: @jason-fries 

Fixes #533

### CODE REVIEW

**Comment by ajratner** [comment]
@henryre I thought there was a TF class for doing this?  Did you check out?

**Comment by henryre** [comment]
That's batch normalization. Will do this for deepier models. This is much more basic.

**Comment by ajratner** [comment]
K will review in a bit

File: snorkel/contrib/learning/relstm.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def train(self, candidates, training_marginals, n_epochs=25, lr=0.01,
             @lr: learning rate
             @dim: embedding dimension
             @batch_size: batch size for mini-batch SGD
-            @rebalance: rebalance training examples?
+            @rebalance: bool or fraction of positive examples desired
```
Comment (ajratner) [inline snorkel/contrib/learning/relstm.py]: Just make this a tiny bit more clear- e.g. state that 0.5 would represent standard class balancing, `False` means no class balancing, what is `True`?

File: snorkel/learning/utils.py [hunk 2]
```diff
@@ -11,6 +11,51 @@
...
+class LabelBalancer(object):
+    def __init__(self, y):
+        """Utility class to rebalance training labels
+        For example, to get the indices of a training set
+        with labels y and around 90 percent negative examples,
+            LabelBalancer(y).get_train_idxs(rebalance=0.1)
+        """
+        self.y = np.ravel(y)
+    
+    def _get_pos(self, split):
```
Comment (ajratner) [inline snorkel/learning/utils.py]: `split` is a weird name for this

File: snorkel/contrib/learning/relstm.py (hunk 1 above)
Comment (henryre) [inline snorkel/contrib/learning/relstm.py]: Updated comment.

File: snorkel/learning/utils.py (hunk 2 above)
Comment (henryre) [inline snorkel/learning/utils.py]: Disagree, and this is an internal method so YOLO.
